src/media/js/views/search.js

```diff
@@ -208,10 +208,18 @@ define('views/search',
             if (query) break;
         }
 
-        if (query && queryParam === 'author') {
-            pageTypes += ' leaf';
-            utilsLocal.headerTitle(gettext('Developer Listing'));
-        } else if (!settings.meowEnabled) {
+        if (query) {
+            switch (queryParam) {
+                case 'author':
+                    pageTypes += ' leaf';
+                    utilsLocal.headerTitle(gettext('Developer Listing'));
+                    break;
+                case 'q':
+                    pageTypes += ' leaf';
+                    utilsLocal.headerTitle(gettext('Search Results'));
+            }
+        }
+        else if (!settings.meowEnabled) {
```

Comment:
Nitpick but our coding style also suggests the `else if ()...` should go one line up after the closing brace (since it's related to the `if (query)` block. This wouldn't prevent an r+ but it's good to be consistent. If @ddurst says the core bug is valid we can probably merge this.